models/repo_collaboration.go

```diff
@@ -279,3 +279,18 @@ func IsOwnerMemberCollaborator(repo *repo_model.Repository, userID int64) (bool,
 
 	return db.GetEngine(db.DefaultContext).Get(&Collaboration{RepoID: repo.ID, UserID: userID})
 }
+
+// GetRepoCollaboratorAccessMode gets the repository access mode for the given user
+func GetRepoCollaboratorAccessMode(repo *repo_model.Repository, userID int64) (perm.AccessMode, error) {
```

Comment:
we dont have a func already for this?!?

Reply:
we can delete this func and use p, e := GetUserRepoPermission(repo, user) directly

Reply:
can't really find one, any suggestion? `GetUserRepoPermission` is used in many places ;)

Reply:
I must have been blind